lib/interfaces/tdd.js

```diff
@@ -99,9 +101,12 @@ module.exports = function(suite){
      */
 
     context.test.only = function(title, fn){
-      var test = context.test(title, fn);
-      var reString = '^' + escapeRe(test.fullTitle()) + '$';
-      mocha.grep(new RegExp(reString));
+      var test = context.test(title, fn)
+        , suite = test.parent;
+      mocha.options.hasOnly = true;
+      suite.isOnly = true;
+      suite.hasOnly = (suite.hasOnly || []).concat(test);
```

Comment:
Isn't a bit confusing that you are using `hasOnly` as a `Boolean` in both `mocha.options` and `Runner`, and then you use it as an `Array` in `Suite`?

Besides that, the variable name `hasOnly` seems appropriate for a `Boolean`, but not so much for an `Array`.

e.g. `suite.tests` is the `Array` with the tests, `suite.onlyTests` could be the `Array` with the tests that 'are only'. It describes much better the content of the property.